src/main/scala/com/amazon/deequ/profiles/ColumnProfiler.scala

```diff
@@ -347,18 +357,31 @@ object ColumnProfiler {
   private[this] def extractGenericStatistics(
       columns: Seq[String],
       schema: StructType,
-      results: AnalyzerContext)
+      results: AnalyzerContext,
+      predefinedColumnDataTypes: Option[Map[String, String]])
     : GenericColumnStatistics = {
 
     val numRecords = results.metricMap
       .collect { case (_: Size, metric: DoubleMetric) => metric.value.get }
       .head
       .toLong
 
-    val inferredTypes = results.metricMap
-      .collect { case (analyzer: DataType, metric: HistogramMetric) =>
-        val typeHistogram = metric.value.get
-        analyzer.column -> DataTypeHistogram.determineType(typeHistogram)
+
+    val inferredTypes =
```

Comment:
This can be simplified. Also, we should not even compute the typehistogram for a column if we do not want to infer its type.

Reply:
Good point, it could be simplified, for now inferredType will only compute typeHistogram for cases when predefinedDataTypes is not defined or when the column is not contained in the keySet of predefinedDataTypes. I also do similar check in typeDetectionHistograms as well.